docs/kb/integrations/defect-dojo-integration.md

```diff
@@ -0,0 +1,114 @@
+
+### DefectDojo API example 
+
+To run API DefectDojo operations such as GET, POST, and DELETE, an API token is necessary. To get it, follow [API guide](https://documentation.defectdojo.com/integrations/api-v2-docs/).
+
+* Once got it, in your system, usually in your CI environment, declare the environment variable `DEFECT_DOJO_API_TOKEN`
+```
+export DEFECT_DOJO_API_TOKEN=[YOUR_DEFECT_DOJO_TOKEN]
+```
+Exploring the DefectDojo API, the endpoint is: `/api/v2/import-scan/` for the first import and `/api/v2/reimport-scan` for the following imports.
+
+These endpoints receive some parameters. The following must be specified:
+* `file`: The Semgrep scan findings report or export in JSON format
+* `scan_type`: A descriptive name for the scan type. In this example, we use "Semgrep JSON Report" for Semgrep importation.
+* `product_name`: The name of the product in DefectDojo to send the Semgrep findings report to.
+* `engagement_name`: Depending on your use case, the [`engagement_name`](https://defectdojo-dev.readthedocs.io/en/latest/about.html#engagements) can describe a moment in time that the test is taking place in, or it can be used as a simple description about the import. In this case, you can name it `semgrep`.
+
+So a Python function using this endpoint can look like this:
+
+```
+def uploadToDefectDojo(is_new_import, token, url, product_name, engagement_name, filename):
+    multipart_form_data = {
+        'file': (filename, open(filename, 'rb')),
+        'scan_type': (None, 'Semgrep JSON Report'),
+        'product_name': (None, product_name),
+        'engagement_name': (None, engagement_name),
+    }
+
+    endpoint = '/api/v2/import-scan/' if is_new_import else '/api/v2/reimport-scan/'
+    r = requests.post(
+        url + endpoint,
+        files=multipart_form_data,
+        headers={
+            'Authorization': 'Token ' + token,
+        }
+    )
+    if r.status_code != 200:
+        sys.exit(f'Post failed: {r.text}')
+    print(r.text)
+```
+
+The full version of this Python script can be found in the [Semgrep Utilities repository](https://github.com/r2c-CSE/semgrep-utilities/blob/main/integrations/defectdojo/import_semgrep_to_defect_dojo.py).
+It is ready to execute.
+
+### Running the script
+
+To run the script:
+```
+python3 integrations/defectdojo/import_semgrep_to_defect_dojo.py --host DOJO_URL --product PRODUCT_NAME --engagement ENGAGEMENT_NAME --report REPORT_FILE 
+```
+Where:
+* `DOJO_URL` is the URL where DefectDojo is.
+* `PRODUCT_NAME` is the DefectDojo product name.
+* `ENGAGEMENT_NAME` is the DefectDojo engagement name for that product.
+* `REPORT_FILE` is the Semgrep report path
+
+## Integrating Semgrep and DefectDojo in a CI pipeline
+
+To prevent tampering at any point of delivery, it is crucial to integrate security scans and dump results to DefectDojo in the **same pipeline or CI job**.
```

Comment:
```suggestion
To prevent tampering of findings at any point of delivery, it is crucial to integrate importing of scan results to DefectDojo in the **same pipeline or CI job** as the scan itself.
```